platform/spec/services/etl/extractors/convert_read_record/incremental_dedup_service_spec.rb

```
@@ -0,0 +1,140 @@
+      let(:record_data) do
+        [
+          { 'id' => 1, 'name' => 'Test 1' },
+          { 'id' => 2, 'name' => 'Test 2' }
+        ]
+      end
+
+      it 'creates write records for new data' do
+        expect { service.call }.to change(SyncWriteRecord, :count).by(2)
+      end
+
+      context 'when duplicate records exist' do
```

rubocop (1.69.1)

[convention] Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

(Style/StringLiterals)